### PR TITLE
test: simulation and CI/CD

### PR DESCRIPTION
**Pull request by poorphd**
## Description

Test PR for simulation and CI/CD 

### Workflow
- [x] build - checking build completed with no errors on PR
- [x] test / codecov - checking test coverage for each commit on PR
- [x] simulations for cosmos-sdk modules

### Simulation
Default simulations for Cosmos-sdk modules were broken. Followings are how we fixed issues:
- not increasing block height - because `sdk. DefaultPowerReduction` in `app.go` was set to [`ethermint.PowerReduction`](https://github.com/b-harvest/Canto/blob/simulation-cicd/app/app.go#L150), Tendermint recognizes the power of the validator as 0, causing the block not to progress. To resolve this issue, we set manual `sdk.DefaultPowerReduction` value in [each simulation test](https://github.com/b-harvest/Canto/blob/simulation-cicd/app/sim_test.go#L252). 
- `NewValidatorCommissionDecorator` in Cosmos anteHandler breaks the simulation - To resolve this issue, SDK modification is required and to avoid this, [a new separated anteHandler](https://github.com/b-harvest/Canto/blob/simulation-cicd/app/ante/handler_options.go#L104-L130) for simulation is introduced.
- `ChainID` - EVM rejects the [invalid ChainID](https://github.com/evmos/ethermint/blob/07cf2bd2b1ce9bdb2e44ec42a39e7239292a14af/types/chain_id.go#L52-L70) and it breaks the simulation. To fix this issue, [proper chain ID is set](https://github.com/b-harvest/Canto/blob/simulation-cicd/app/sim_test.go#L190-L191) when creating a new `ctx`.
 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] targeted the correct branch (see [PR Targeting](https://github.com/canto-network/canto/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] included the necessary unit and integration tests
- [ ] reviewed "Files changed" and left comments if necessary
 
### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by poorphd** [comment]
have verified that the existing cosmos module simulation is working properly as of this branch.

File: app/ante/handler_options.go [hunk 1]
```diff
@@ -25,12 +26,14 @@ type HandlerOptions struct {
 	BankKeeper      evmtypes.BankKeeper
 	IBCKeeper       *ibckeeper.Keeper
 	FeeMarketKeeper evmtypes.FeeMarketKeeper
+	SlashingKeeper  *slashingkeeper.Keeper
```
Comment (dongsam) [inline app/ante/handler_options.go:29]: This is what `NewParamChangeLimitDecorator` needs in liquidstaking(WIP) and I don't think I need it now

Comment (poorphd) [reply app/ante/handler_options.go]: I've removed it. https://github.com/b-harvest/Canto/pull/23/commits/9c100e5412c6982c89f2a662ab2c07d8d615ddfd

Comment (poorphd) [reply app/ante/handler_options.go]: It's removed (https://github.com/b-harvest/Canto/pull/23/commits/a719d632c2f3b5f0afb2378df07fe4b9c3b81bfd)

**Comment by dongsam** [comment]
Updated the target module for simulation and Refactor to recycle the structure in simapp @poorphd

**Pull request merged**